From 2a83121d4b0ad835c710b10778dd0a8c89ca74ff Mon Sep 17 00:00:00 2001
From: Jiaxun Yang <jiaxun.yang@flygoat.com>
Date: Sat, 7 Jan 2023 22:59:27 +0000
Subject: [PATCH 138/138] MIPS: Loongson64: Fix NUMA distance calculation

Use LOCAL_DISTANCE to define LOCAL_DISTANCE and reduce remote
node distance to 100 to avoid exceeding distance range.

Signed-off-by: Jiaxun Yang <jiaxun.yang@flygoat.com>
---
 arch/mips/loongson64/loongson-3/numa.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/mips/loongson64/loongson-3/numa.c b/arch/mips/loongson64/loongson-3/numa.c
index 6402da80f..89989278a 100644
--- a/arch/mips/loongson64/loongson-3/numa.c
+++ b/arch/mips/loongson64/loongson-3/numa.c
@@ -128,7 +128,7 @@ static void cpu_node_probe(void)
 		loongson_sysconf.nr_cpus, num_online_nodes());
 }
 
-static int __init compute_node_distance(int row, int col)
+static unsigned char __init compute_node_distance(int row, int col)
 {
 	int package_row = row * loongson_sysconf.cores_per_node /
 				loongson_sysconf.cores_per_package;
@@ -136,11 +136,11 @@ static int __init compute_node_distance(int row, int col)
 				loongson_sysconf.cores_per_package;
 
 	if (col == row)
-		return 0;
+		return LOCAL_DISTANCE;
 	else if (package_row == package_col)
 		return 40;
 	else
-		return 200;
+		return 100;
 }
 
 static void __init init_topology_matrix(void)
-- 
2.39.1

